### PR TITLE
decorate a class

### PR DESCRIPTION
**Pull request by fickryiman**
- Implement composition (as an example of the design pattern)
- Create a class Nameable
- Person inherits from Nameable class
- Create class Decorator
- Decorator inherits from Nameable class
- Create class CapitalizeDecorator
- CapitalizeDecorator inherits from Decorator class
- Create class TrimmerDecorator
- TrimmerDecorator inherits from Decorator class
- create main.rb to Run a program using the command line.

### CODE REVIEW

File: class/trimmer_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+require_relative 'decorator'
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    trimmed = @nameable.correct_name[0, 9]
+    @nameable.correct_name.length >= 10 ? trimmed : @nameable.correct_name
+  end
```
Comment (KanzaTahreem) [inline class/trimmer_decorator.rb]: - [x] Great job for implementing the `TrimmerDecorator` 👍👍 However, please make sure to return maximum of 10 characters not 9 as per [this requirement](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_decorator_v2.md#:~:text=Implement%20a%20method%20correct_name%20that%20makes%20sure%20that%20the%20output%20of%20%40nameable.correct_name%20has%20a%20maximum%20of%2010%20characters.%20If%20it%27s%20longer%20it%20should%20trim%20the%20word.). To fix this, you can use the following 👇✨

```ruby
 trimmed = @nameable.correct_name[0, 10]
 # or
 trimmed = @nameable.correct_name[0..9]
```

**_current implementation_**

image

**Review by KanzaTahreem — CHANGES_REQUESTED** [review]
# Changes Required 🔁

Hi @fickryiman 

Good job so far👏! I really appreciate the effort you put in!  **_However, there are few this that still need to be fixed. But you are almost there!_** 💪 🏁

so close


## Highlights 🎯

There are some excellent points that you have made perfectly. Keep up the great work! 👏
- ✅ Good job with the decorators
- ✅ Descriptive PR and README file
- ✅ Correct Linters

## Required Changes 🛑

_Please check the comments under the review._

## Optional suggestions ❕

N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: class/trimmer_decorator.rb (hunk 1 above)
Comment (fickryiman) [reply class/trimmer_decorator.rb]: Hi @KanzaTahreem,

Fixes the the trimmed name to return 10 characters..

**Review by Shedrack-Sunday — APPROVED** [review]
Hi @fickryiman 

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

image


Highlights
- [x] Good project organization ✔️
- [x] Nice use of Ruby features ✔️ 
- [x] Kudus on meeting the requirements suggested  ✔️



Optional suggestions
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**